extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/cors/CORSFilter.java

```diff
@@ -24,9 +24,10 @@ private void processRequestedHeaders(HttpServerResponse response, String request
         if (corsConfig.headers.isEmpty()) {
             response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, requestedHeaders);
         } else {
+            String lcRequestedHeaders = requestedHeaders.toLowerCase();
             final String validRequestHeaders = corsConfig.headers
                     .stream()
-                    .filter(requestedHeaders::contains)
+                    .filter(configHeader -> lcRequestedHeaders.contains(configHeader.toLowerCase()))
                     .collect(Collectors.joining(","));
```

Comment:
@gsmet Is it OK to keep this lambda here?

Reply:
Should we lower case requested headers too?

Reply:
We should and it's already done at line 27 :)

Reply:
Oops, my bad. Please Ignore my comment, I missed that part. Thanks :-)

Reply:
I think I would have lowercased the config headers at the init of the filter once and for all.

As it's runtime code and the lambda is not that useful, I would have removed it.

And, if I'm not mistaken, the current code is fragile and unsafe. I think we should split the requested headers before doing any comparison. Otherwise, you could have a partial match.

And I would probably have done the same for the methods.

Reply:
I did lowercase the headers in the filter constructor at first and then saw that `CORSConfig` was part of `HttpConfiguration` which is annotated with `@ConfigRoot(phase = ConfigPhase.RUN_TIME)`. Knowing that, is it really OK to lowercase once at the filter init?

Reply:
Hmmm, you’re right.

Reply:
Yup, that's exactly why I'm not happy with the current code.

Reply:
And it's about to be changed, I'll update the PR as soon as I find some time :)